torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -203,6 +211,12 @@ inline Tensor as_view(const Tensor & base, const Tensor & tensor, bool is_bw_dif
 // See NOTE [ Autograd View Variables ] for details.
 inline std::vector<Tensor> as_view(const Tensor & base, std::vector<Tensor>& tensors, bool is_bw_differentiable,
                                    bool is_fw_differentiable, CreationMeta creation_meta=CreationMeta::DEFAULT) {
+  // See Note [View of inference tensor]
+  if (base.unsafeGetTensorImpl()->is_inference_tensor() ||
+        std::all_of(
```

Comment:
Why is this all and not any to match the or above?

Reply:
This `std::all` will be removed as we've enforce base and tensor are either both inference or normal. 
I had this check as all mainly because all tensors should have been created at the same time (thus of the same type).